code/game/objects/structures/aliens.dm

```diff
@@ -126,6 +126,8 @@
 	canSmoothWith = list(/obj/structure/alien/weeds, /turf/closed/wall)
 	smooth = SMOOTH_MORE
 
+var/list/blacklisted_weed_turfs = typecacheof(list(/turf/open/floor/plating/lava))
```

Comment:
make this ```/static``` on the type, and drop "weed" (since it won't be needed when it's on the type)